inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -147,16 +148,17 @@ def formdata_to_model(obj, formdata):
         data['collections'].extend([{'primary': "ConferencePaper"}])
 
     # ============================
-    # Page range
+    # Page number
     # ============================
     if 'page_nr' not in data:
-        if data.get("publication_info", [{}])[0].get("page_artid"):
-            pages = data['publication_info'][0]['page_artid'].split('-')
-            if len(pages) == 2:
-                try:
-                    data['page_nr'] = int(pages[1]) - int(pages[0]) + 1
-                except ValueError:
-                    pass
+        first_publication_info = data.get('publication_info', [{}])[0]
+
+        page_start = first_publication_info.get('page_start')
+        page_end = first_publication_info.get('page_end')
+
+        if page_start and page_end:
+            data['page_nr'] = page_end - page_start + 1
```

Comment:
Ouch, I was wrong here. I was assuming `page_start` and `page_end` were integers from the schema, but they are not, so the difference will throw a `TypeError` here. Which means in fact this should be:
```python
try:
    data['page_nr'] = int(page_end) - int(page_start) + 1
except (TypeError, ValueError):
    pass
```